rfcs/0038-rosetta-construction-api.md

```diff
@@ -0,0 +1,424 @@
+                             X                                       |
+                             X Sign Payload(s) +-----------------------------> /construction/combine
+                             X (using caller's own detached signer)  |                 +
+                             X                                       |                 |
+   Create signed transaction X         +-----------------------------------------------+
+                             X         v                             |
+                             X Parse Signed Transaction +--------------------> /construction/parse
+                             X to Confirm Correctness                |
+                             X                                       |
+                                                                     |
+                             X                                       |
+                             X Get hash of signed transaction +--------------> /construction/hash
+Broadcast Signed Transaction X to monitor status                     |
+                             X                                       |
+                             X Submit Transaction +--------------------------> /construction/submit (online)
+                             X                                       |
+                                                                     +
+```
+
+This flow chart will guide our explanation of what is needed to build a full to-spec implementation of the API. Afterwards, we'll list out each proposed piece of work with details of how it should be implemented. Upon mergin of this RFC, each work item will become an issue on GitHub.
+
+The initial version of the construction API will _only_ support payments. We can quickly follow with a version that will support delgation as well. This RFC will talk about all the tasks assuming both payments and delegations are supported. All token-specific transactions (and any future transactions) are not yet supported and out-of-scope for this RFC.
+
+### Flow chart
+
+#### Before Derivation
+
+Before the derivation step, we need to generate a keypair. We'll use the private key to sign the transaction and the public key to tell others who the sender of the transaction is.
+
+#### Derivation
+
+[derivation]: #derivation
+
+Derivation demands that the public key expected as input be a hex-encoded byte-array value. So we'll [add functionality](#marshalkeys) to the [client-sdk](#marshalkeys), the [generate-keypair binary](#marshalkeys), and the offcial [Coda CLI](#marshalkeys) to marshall the `Fq.t * Fq.t` pair (the native representation of an uncompressed public key).
+
+The [derivation endpoint](#derivation-endpoint) would be responsible for reading in the uncompressed public key bytes which [requires adjusting the Rosetta spec](#addcurves), compressing the public key, and base58-encoding it inline with how we currently represent public keys in serialized form.
```

Comment:
I don't understand how the result is used. Coinbase sends a byte array, which we'd convert to `Public_key.Compressed.t`, which is a Coda-internal format.

Why the additional conversion to Base58Check? Is that the format used for later steps?

Reply:
> Why the additional conversion to Base58Check? Is that the format used for later steps?

Yes, it's (1) the format used for later steps and (2) the format we use for our "human readable" addresses.